mir_eval/sonify.py

```diff
@@ -102,17 +102,24 @@ def time_frequency(gram, frequencies, times, fs, function=np.sin, length=None):
     times, _ = util.adjust_intervals(times, t_max=length)
 
     # Truncate times so that the shape matches gram
-    times = times[:gram.shape[1]]
+    n_times = gram.shape[1]
+    times = times[:n_times]
 
     def _fast_synthesize(frequency):
         """A faster way to synthesize a signal.
             Generate one cycle, and simulate arbitrary repetitions
             using array indexing tricks.
         """
-        frequency = float(frequency)
+        # hack so that we can ensure an integer number of periods and samples
+        # rounds frequency to 1st decimal, s.t. 10 * frequency will be an int
+        frequency = np.round(frequency, 1)
```

Comment:
offline conversation summary: might be better to make the number of decimals a parameter `n_dec`, and maintain a table of size `10**n_dec * fs` so that a user can control the precision of the approximation.